tls/s2n_client_key_exchange.c

```diff
@@ -209,6 +239,36 @@ int s2n_kem_client_key_send(struct s2n_connection *conn, struct s2n_blob *shared
     return 0;
 }
 
+int s2n_hybrid_client_key_send(struct s2n_connection *conn, struct s2n_blob *combined_shared_key)
```

Comment:
There is a lot of repetition between send and recv. Could you do this by taking a function pointer to the IO operation passing in s2n_kex_client_key_X function or did I miss other small differences?

Reply:
I agree there is a lot of duplication. There are two small differences: `s2n_kex_client_key_send` vs `s2n_kex_client_key_recv` and using the read or write cursor to get the client key exchange. 

For the key_recv vs key_send thats easy to extract another level of indirection. For the cursor I could check if the read or write cursor is changing to infer what side is calling the method. Or add another function pointer to get the correct difference.